internal/healthcheck/healthcheck.go

```diff
@@ -64,6 +85,13 @@ func HealthChecks(ctx context.Context, c client.Reader, l logr.Logger) error {
 		failures = append(failures, err.Error())
 	}
 
+	if !opts.SkipSubscriptionCheck {
```

Comment:
I thought subscription check were useful in general. The only bit that may need to be skipped is the upgradePending case?

Reply:
Oh, never mind. I thought you were checking `state: UpgradePending` vs `state: AtLatestKnown`

Reply:
Subscription checks have been tightened to only consider the Failed and CatalogSourcesUnhealthy conditions